demos/quiz_bot/screens.py

```diff
@@ -0,0 +1,249 @@
+        return [[
+            Button(
+                _('⬅️ Back', language_code),
+                MainMenuScreen,
+                source_type=SourceTypes.MOVE_SOURCE_TYPE),
+        ]]
+
+    async def get_initial_value(self, _update, context):
+        """Return updated or default language code."""
+        language_code = context.user_data.get('language_code', 'en')
+        if not language_code:
+            return 'en'
+
+        return language_code
+
+    async def switch(self, update, context, selected_choice):
+        """Save the selected language and re-render the screen."""
+        selected_language, _ = selected_choice
+        context.user_data['language_code'] = selected_language
+
+        return await super().switch(update, context, selected_choice)
+
+
+class MainMenuScreen(BaseScreen, StartMixin):
+    """The class implements MainMenuScreen."""
+
+    caption = 'main_menu_screen'
+
+    async def add_default_keyboard(self, _update, context):
+        """Set up the default keyboard for the screen."""
+        language_code = context.user_data.get('language_code', 'en')
+        return [
+            [Button(
+                _('❓ Start Quiz', language_code),
+                self.start_quiz_handler,
+                source_type=SourceTypes.HANDLER_SOURCE_TYPE,
+            )],
+            [Button(
+                _('🌍 Change Language', language_code),
+                LanguageSwitcherScreen,
+                source_type=SourceTypes.MOVE_SOURCE_TYPE,
+            )],
+        ]
+
+    async def render(self, update, context, *, config=None, **kwargs):
+        """Initiate or reset necessary data and set new questions for the quiz."""
+        context.user_data['question_index'] = \
+        context.user_data['correct_answers_num'] = \
+        context.user_data['answer'] = 0
+
+        context.user_data['questions'] = random.sample(context.user_data['all_questions'], k=5)
+
+        return await super().render(update, context, config=config, **kwargs)
+
+    async def start(self, update, context):
+        """Reply to the /start command, load and save the questions from the file."""
+        async with aiofiles.open(settings.BASE_DIR / 'questions.json') as file:
+            questions = json.loads(await file.read())
+            context.user_data['all_questions'] = questions
+
+        return await super().start(update, context)
+
+    @register_button_handler
+    async def start_quiz_handler(self, update, context):
+        """Handle a request to start the quiz."""
+        index = context.user_data['question_index']
+        questions = context.user_data['questions']
+
+        return await self.get_next_choice_widget(
+            questions[index]['correct_answer'],
+        )().move(update, context)
+
+
+class QuizMultiChoiceWidget(BaseQuizScreen, MultiChoiceWidget):
+    """The class implements QuizSingleChoiceWidget, which acts as a response
+    if the question has more than one correct answer.
+    """
```

Comment:
`The class implements QuizSingleChoiceWidget, which is used when a question has more than one correct answer.`